apps/webapp/app/routes/resources.environments.$environmentId.events.$eventId.cancel.ts

```diff
@@ -0,0 +1,62 @@
+import { parse } from "@conform-to/zod";
+import { ActionFunction, json } from "@remix-run/node";
+import { z } from "zod";
+import { redirectWithSuccessMessage } from "~/models/message.server";
+import { logger } from "~/services/logger.server";
+import { CancelEventService } from "~/services/events/cancelEvent.server";
+import { prisma } from "~/db.server";
+
+export const cancelEventSchema = z.object({
+  redirectUrl: z.string(),
+});
+
+const ParamSchema = z.object({
+  environmentId: z.string(),
+  eventId: z.string(),
+});
+
+export const action: ActionFunction = async ({ request, params }) => {
+  const { environmentId, eventId } = ParamSchema.parse(params);
```

Comment:
We should `requireUserId` and then lower down check that the userId is allowed to cancel the event.